src/index.js

```diff
@@ -1,32 +1,43 @@
 var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
+var API = "https://rickandmortyapi.com/api/character/";
 var xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+var fetchData = function(url_api, data) {
+  return new Promise(function(resolve, reject) {
+    xhttp.onreadystatechange = function() {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          // If successful
+          resolve(xhttp);
+        } else {
+          // If failed
+          reject({
+            status: request.status,
+            statusJSON: request.statusText
+          });
+        }
+      }
+    };
+    xhttp.open("GET", url_api, false);
+    xhttp.send();
+  });
 };
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
+fetchData(API)
+  .then(data => {
+    console.log("Primer Llamado...");
+    data = JSON.parse(data.responseText);
+    console.log(`Personajes: ${data.info.count}`);
+    fetchData(API + data.results[0].id).then(data => {
+      data = JSON.parse(data.responseText);
+      console.log(`Primer Personaje: ${data.name}`);
+      fetchData(data.origin.url).then(data => {
+        data = JSON.parse(data.responseText);
+        console.log(`Dimensión: ${data.dimension}`);
+      });
     });
+  })
+  .catch(function(error) {
```

Comment:
Utilicemos Arrow Functions. 

```suggestion
  .catch((error) => {
```